src/main/java/ca/on/oicr/pineryreports/util/SampleUtils.java

```diff
@@ -225,6 +226,26 @@ public static String getUpstreamField(Function<SampleDto, String> getField, Samp
 	return null;
   }
   
+  /**
+   * Get an attribute from lower in the hierarchy. Will never return an attribute found directly on sample.
+   * 
+   * @param attributeName
+   * @param sample find the attribute in this sample's children
+   * @param possibleChildren set of potential children to this sample
+   * @return the attribute value set from the children which have the attribute, or empty set if not found
+   */
+  public static Set<String> getChildAttributes(String attributeName, SampleDto sample, List<SampleDto> possibleChildren) {
+    Set<String> foundAttributes = new HashSet<>();
+    for (SampleDto child : possibleChildren) {
+      if (getParentId(child) == null || !getParentId(child).equals(sample.getId())) continue;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "getParentId()" can return null. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2259)